src/main/java/core/basesyntax/strategy/BalanceOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int quantity = fruitTransaction.getQuantity();
+        Storage.fruitStorage.put(fruit, quantity);
+    }
```

Comment:
The `apply` method should check if the quantity is positive, as negative balances are not valid for the 'balance' operation. Throw a `RuntimeException` if the quantity is negative.